src/pkg/services/m365/api/channels_pager.go

```diff
@@ -14,30 +21,205 @@ type ChannelMessageDeltaEnumerator interface {
+}
+
+func (c Channels) NewChannelPager(
+	teamID,
+	channelID string,
+	fields []string,
+) *channelPageCtrl {
+	requestConfig := &teams.ItemChannelsRequestBuilderGetRequestConfiguration{
+		QueryParameters: &teams.ItemChannelsRequestBuilderGetQueryParameters{
+			Select: fields,
+		},
+	}
+
+	res := &channelPageCtrl{
+		gs:      c.Stable,
+		options: requestConfig,
+		builder: c.Stable.
+			Client().
+			Teams().
+			ByTeamId(teamID).
+			Channels(),
+	}
 
-// type channelsPageCtrl struct {
-// 	gs      graph.Servicer
-// 	builder *teams.ItemChannelsChannelItemRequestBuilder
-// 	options *teams.ItemChannelsChannelItemRequestBuilderGetRequestConfiguration
-// }
+	return res
+}
+
+func (p *channelPageCtrl) SetNext(nextLink string) {
+	p.builder = teams.NewItemChannelsRequestBuilder(nextLink, p.gs.Adapter())
+}
+
+func (p *channelPageCtrl) GetPage(ctx context.Context) (PageLinker, error) {
+	var (
+		resp PageLinker
+		err  error
+	)
+
+	resp, err = p.builder.Get(ctx, p.options)
+	if err != nil {
+		return nil, graph.Stack(ctx, err)
+	}
+
+	return resp, nil
+}
+
+func (p *channelPageCtrl) ValuesIn(l PageLinker) ([]models.Channelable, error) {
+	return getValues[models.Channelable](l)
+}
+
+func (p *channelPageCtrl) GetOdataNextLink() *string {
+	// No next link preent in the API result
+	emptyString := ""
+	return &emptyString
+}
+
+// ---------------------------------------------------------------------------
+// non delta item pager
+// ---------------------------------------------------------------------------
+
+type MessageItemIDType struct {
+	ItemID string
+}
+
+type channelItemPageCtrl struct {
+	gs      graph.Servicer
+	builder *teams.ItemChannelsItemMessagesRequestBuilder
+	options *teams.ItemChannelsItemMessagesRequestBuilderGetRequestConfiguration
+}
+
+func (c Channels) GetItemIDsInContainer(
+	ctx context.Context,
+	teamID, channelID string,
+) (map[string]MessageItemIDType, error) {
+	ctx = clues.Add(ctx, "channel_id", channelID)
+	pager := c.NewChannelItemPager(teamID, channelID)
+
+	items, err := enumerateItems(ctx, pager)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "enumerating contacts")
+	}
+
+	m := map[string]MessageItemIDType{}
+
+	for _, item := range items {
+		m[ptr.Val(item.GetId())] = MessageItemIDType{
+			ItemID: ptr.Val(item.GetId()),
+		}
+	}
+
+	return m, nil
+}
+
+func (c Channels) NewChannelItemPager(
+	teamID, containerID string,
+	selectProps ...string,
+) itemPager[models.ChatMessageable] {
+	options := &teams.ItemChannelsItemMessagesRequestBuilderGetRequestConfiguration{
+		QueryParameters: &teams.ItemChannelsItemMessagesRequestBuilderGetQueryParameters{},
+	}
+
+	if len(selectProps) > 0 {
+		options.QueryParameters.Select = selectProps
+	}
+
+	builder := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Messages()
+
+	return &channelItemPageCtrl{c.Stable, builder, options}
+}
+
+//lint:ignore U1000 False Positive
```

Comment:
is this necessary, or can it be removed?

Reply:
Yes. Its required. Linter fails because of it.